gcp/cluster.tf

```diff
@@ -119,12 +122,36 @@ KEYS
   labels = tomap(var.labels)
 }
 
+locals {
+  hosts_map = {
+    for i in range(length(var.availability_zone)) : var.availability_zone[i] =>
+    concat(
+      [
+        for j in range(i, length(google_compute_instance.redpanda), length(var.availability_zone)) :
+        google_compute_instance.redpanda[j].self_link
+      ],
+      [
+        for j in range(i, length(google_compute_instance.client), length(var.availability_zone)) :
+        google_compute_instance.client[j].self_link
+      ],
+      [
+        for j in range(i, length(google_compute_instance.monitor), length(var.availability_zone)) :
+        google_compute_instance.monitor[j].self_link
+      ]
+    )
+
+  }
+}
+
```

Comment:
Can you add a comment here to describe what this is doing?

Reply:
I'm honestly struggling to understand what this is doing and why it is doing it. I'd have to run it to be sure, but it kinda looks like it is generating self links for each az for each instance even when some of the things (monitor, client) should only be created in one az.

Reply:
```suggestion
// generate a list of hosts (redpanda instances, clients, and monitor instances) for each availability zone
locals {
  hosts_map = {
    for i in range(length(var.availability_zone)) : var.availability_zone[i] =>
    concat(
      [
        for j in range(i, length(google_compute_instance.redpanda), length(var.availability_zone)) :
        google_compute_instance.redpanda[j].self_link
      ],
      [
        for j in range(i, length(google_compute_instance.client), length(var.availability_zone)) :
        google_compute_instance.client[j].self_link
      ],
      [
        for j in range(i, length(google_compute_instance.monitor), length(var.availability_zone)) :
        google_compute_instance.monitor[j].self_link
      ]
    )

  }
}

```

Reply:
Yes, so it's generating a map of AZ-> Host which is then used to generate the instances section of the google_compute_instance_group. Previously we just enumerated all of the brokers, clients and monitoring nodes, but because we could have n instance_groups we now need to be able to iterate across the hosts on a per-AZ basis.

Reply:
Added a comment explaining this and why we do it